File: manuscript/Ch06 Application Development/Examples/golang/pools.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/jackc/pgx/pgxpool"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Fprintln(os.Stderr, "Missing URL argument")
		os.Exit(1)
	}
	uri := os.Args[1]
	ctx := context.Background()
	config, err := pgxpool.ParseConfig(uri)
	config.MaxConns = 40
	pool, err := pgxpool.ConnectConfig(ctx, config)
	defer pool.Close()

	connection, err := pool.Acquire(ctx)

	result, err := connection.Query(context.Background(), "SELECT NOW()")
	if err != nil {
		fmt.Fprintf(os.Stderr, "CockroachDB error: %v\n", err)
	}
	defer result.Close()
	result.Next()
	values, err := result.Values()
	if err != nil {
		panic(err)
	}
	fmt.Fprintf(os.Stdout, "%v \n", values[0])

}

File: manuscript/Ch06 Application Development/Examples/python/pools.py
#!/usr/bin/env python3


import sys
import time
import csv

import psycopg2
from psycopg2 import pool
 
def main():

  if ((len(sys.argv)) !=2):
    sys.exit("Error:No URL provided on command line")
  uri=sys.argv[1]

  pool= psycopg2.pool.ThreadedConnectionPool(10, 40, uri) # min connection=10, max=40

  connection = pool.getconn()

  cursor=connection.cursor()
  cursor.execute("SELECT now()")
  for row in cursor:
    print(row[0])
  cursor.close()
  connection.close()
 
main()